orttraining/orttraining/test/training_api/checkpoint/checkpoint_test.cc

```diff
@@ -0,0 +1,328 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "orttraining/core/framework/checkpointing.h"
+
+#include <unordered_map>
+#include <vector>
+
+#include "gtest/gtest.h"
+
+#include "core/common/common.h"
+#include "core/common/logging/logging.h"
+#include "core/common/logging/sinks/clog_sink.h"
+#include "core/framework/data_transfer.h"
+#include "core/framework/ort_value.h"
+#include "core/framework/tensor.h"
+#include "core/framework/tensorprotoutils.h"
+#include "core/graph/graph_viewer.h"
+#include "core/graph/model.h"
+#include "core/platform/path_lib.h"
+
+#include "test/test_environment.h"
+#include "test/util/include/asserts.h"
+#include "test/util/include/temp_dir.h"
+#include "test/util/include/test/test_environment.h"
+
+#include "orttraining/training_api/checkpoint.h"
+#include "orttraining/training_api/utilities.h"
+#include "orttraining/training_api/interfaces.h"
+
+using onnxruntime::test::TemporaryDirectory;
+using namespace onnxruntime::training::api_test;
+namespace onnxruntime {
+namespace training {
+namespace test {
+namespace training_api {
+
+#define MODEL_FOLDER ORT_TSTR("testdata/")
+
+TEST(CheckpointApiTest, SaveOnnxModelAsCheckpoint_ThenLoad_CPU) {
```

Comment:
May we add a description to the test in comments (as in what it is doing)?
Also, it would be much easier to follow the test if we can break it down into three sections (GIVEN, WHEN, THEN) and highlight the sections as comments accordingly. This will help to maintain the tests going forward.

Reply:
I agrre with "May we add a description to the test in comments (as in what it is doing)?", will add one for the test function.

For the next comment, the exiting comment in code is already separating sections, I don't feel we need add more sections.

Reply:
But anyway, adding more comment for explaination.